fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/OnlineIndexBaseBuilder.java

```diff
@@ -0,0 +1,724 @@
+/*
+ * OnlineIndexerBaseBuilder.java
+ *
+ * This source file is part of the FoundationDB open source project
+ *
+ * Copyright 2015-2024 Apple Inc. and the FoundationDB project authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.apple.foundationdb.record.provider.foundationdb;
+
+import com.apple.foundationdb.record.RecordCoreException;
+import com.apple.foundationdb.record.RecordMetaData;
+import com.apple.foundationdb.record.RecordMetaDataProvider;
+import com.apple.foundationdb.record.metadata.Index;
+import com.apple.foundationdb.record.metadata.Key;
+import com.apple.foundationdb.record.metadata.MetaDataException;
+import com.apple.foundationdb.record.provider.common.RecordSerializer;
+import com.apple.foundationdb.subspace.Subspace;
+import com.google.protobuf.Message;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Map;
+import java.util.function.UnaryOperator;
+
+/**
+ * Base class that the {@link OnlineIndexer.Builder} and the {@link OnlineIndexScrubber.Builder} can both inherit
+ * from. Building and scrubbing indexes contains a fair amount of common configuration. This consolidates the
+ * common configuration. Note that each method returns the type specified as the concrete type of the resulting
+ * builder, so each implementation should be able to use this class fluently without casting.
+ *
+ * @param <B> concrete type of this builder
+ */
+public abstract class OnlineIndexBaseBuilder<B extends OnlineIndexBaseBuilder<B>> {
```

Comment:
Can the class be renamed to ` OnlineIndexingBaseBuilder`? 
(I am trying to keep a clear distinction between "index" and "indexing" by avoiding the term "to index" (not always successfully)..)

Reply:
That actually was my initial name. I can change if it you want, but the idea was that this is the basis for both the `OnlineIndexer.Builder` and the `OnlineIndexScrubber.Builder` classes, so I wanted to encapsulate this works for both "online index" operations. I could go with `OnlineIndexOperationBaseBuilder`?